src/com/stanimiriliev/chaptertwo/homogeneoustree/Tree.java

```diff
@@ -1,56 +1,50 @@
 package com.stanimiriliev.chaptertwo.homogeneoustree;
 
 /**
- * Binary tree
- * Created by Stanimir Iliev on 18.6.2017 y..
+ * Binary Ordered Tree, Homogeneous Tree
+ * @author  Stanimir
+ * @since 18.6.2017
  */
 public class Tree {
```

Comment:
I think it will look better if you have class Tree that contains Node root and the methods that are responsible for check(contains), add, print,
Like that you will not have firstAdd  - you will ha ve root element

The node class can look like 
```java
class Node {
  public final Integer value;
  public Node left;
  public Node right;

 public Node(Integer value){
  this.value = value;

public class Tree {
   private Node root;
   public void add(int value){
    ...
   }
   public void print(){
    ..
   }
}
...

also you can define the Node class in as a inner class, because you will use it only in this class, For the Test(App) class you will see and use only the Tree object

 


}

```

For the next task where you should pass different object, you just can change the value that will be in the node -(Object, Comparable)